contracts/contracts/strategies/ThreePoolStrategy.sol

```diff
@@ -0,0 +1,256 @@
+
+    /**
+     * @dev Deposit asset into the Curve 3Pool
+     * @param _asset Address of asset to deposit
+     * @param _amount Amount of asset to deposit
+     * @return amountDeposited Amount of asset that was deposited
+     */
+    function deposit(address _asset, uint256 _amount)
+        external
+        onlyVault
+        returns (uint256 amountDeposited)
+    {
+        require(_amount > 0, "Must deposit something");
+        // 3Pool requires passing deposit amounts for all 3 assets, set to 0 for
+        // all
+        uint256[] memory _amounts = new uint256[](3);
+        // Set the amount on the asset we want to deposit
+        _amounts[uint256(poolCoinIndex)] = _amount;
+        // Do the deposit to 3pool
+        ICurvePool(platformAddress).add_liquidity(_amounts, uint256(0));
+        // Deposit into Gauge
+        IERC20 pToken = IERC20(assetToPToken[_asset]);
+        ICurveGauge(crvGaugeAddress).deposit(
+            pToken.balanceOf(address(this)),
+            address(this)
+        );
+        amountDeposited = _amount;
+        emit Deposit(_asset, address(platformAddress), amountDeposited);
+    }
+
+    /**
+     * @dev Withdraw asset from Curve 3Pool
+     * @param _recipient Address to receive withdrawn asset
+     * @param _asset Address of asset to withdraw
+     * @param _amount Amount of asset to withdraw
+     * @return amountWithdrawn Amount of asset that was withdrawn
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external onlyVault returns (uint256 amountWithdrawn) {
+        require(_recipient != address(0), "Invalid recipient");
+        require(_amount > 0, "Invalid amount");
+        // Calculate how much of the pool token we need to withdraw
+        (
+            uint256 contractPTokens,
+            uint256 gaugePTokens,
+            uint256 totalPTokens
+        ) = _getTotalPTokens();
+        // Calculate the max amount of the asset we'd get if we withdrew all the
+        // platform tokens
+        ICurvePool curvePool = ICurvePool(platformAddress);
+        uint256 maxAmount = curvePool.calc_withdraw_one_coin(
+            totalPTokens,
+            poolCoinIndex
+        );
+        // Calculate how many platform tokens we need to withdraw the asset amount
+        uint256 withdrawPTokens = totalPTokens.mul(_amount).div(maxAmount);
+        if (contractPTokens < withdrawPTokens) {
+            // Not enough of pool token exists on this contract, must be staked
+            // in Gauge, unstake
+            ICurveGauge(crvGaugeAddress).withdraw(withdrawPTokens);
+        }
+        curvePool.remove_liquidity_one_coin(withdrawPTokens, poolCoinIndex, 0);
+        IERC20(_asset).transfer(_recipient, _amount);
+        // Transfer any leftover dust back to the vault buffer.
+        uint256 dust = IERC20(_asset).balanceOf(address(this));
+        if (dust > 0) {
+            IERC20(_asset).safeTransfer(vaultAddress, dust);
```

Comment:
why safeTransfer sometimes used and plain transfer otherwise used?

Reply:
SafeTransfer is only needed if the coin could fail to throw on a failed transfer, such as USDT. So when working with only a specify known good coin, it's fine to just use transfer.

However, this is potentially transferring USDT here, and so should definitely be safeTransfer. Probably my mistake.

^ @tomlinton